src/terrain/clients/iplant_groups.clj

```diff
@@ -259,25 +259,28 @@
 (defn get-communities [user params]
   (get-teams* group-type-communities user params))
 
-(defn- grant-initial-team-privileges [client user group public-privileges]
+(defn- grant-initial-team-privileges [client user group initial-admin public-privileges]
   (c/update-group-privileges client user group
-                             {:updates [{:subject_id (config/grouper-user) :privileges ["admin"]}
+                             {:updates [{:subject_id initial-admin :privileges ["admin"]}
                                         {:subject_id c/public-user :privileges public-privileges}]}))
 
-(defn- add-team* [team-type user {:keys [name description public_privileges] :or {public_privileges []}}]
+(defn add-team [user {:keys [name description public_privileges] :or {public_privileges []}}]
   (let [client (get-client)
-        folder (get-team-folder-name client team-type user)]
-    (ensure-team-folder-exists client team-type user)
-    (let [full-name (str folder ":" name)
+        folder (get-team-folder-name client group-type-teams user)]
+    (ensure-team-folder-exists client group-type-teams user)
+    (let [full-name (full-group-name name folder)
           group     (c/add-group client user full-name team-group-type description)]
-      (grant-initial-team-privileges client user full-name public_privileges)
-      (format-group (get-team-folder-name client team-type) group))))
+      (grant-initial-team-privileges client user full-name (config/grouper-user) public_privileges)
+      (format-group (get-team-folder-name client group-type-teams) group))))
 
-(defn add-team [user request]
-  (add-team* group-type-teams user request))
-
-(defn add-community [user request]
-  (add-team* group-type-communities user request))
+(defn add-community [user {:keys [name description public_privileges] :or {public_privileges []}}]
+  (let [client (get-client)
+        folder (get-team-folder-name client group-type-communities)]
+    (ensure-folder-exists client (config/grouper-user) folder)
+    (let [full-name (full-group-name name folder)
+          group     (c/add-group client (config/grouper-user) full-name team-group-type description)]
+      (grant-initial-team-privileges client (config/grouper-user) full-name user public_privileges)
+      (format-group folder group))))
```

Comment:
Other than the point that Ian brought up, my only concern would be that we'd need to make sure that users can rename and delete communities that they administer. I don't think that Grouper privileges will prevent that, but I don't remember for certain.

Reply:
Good point. I just tested it, and it seems that admins can delete the community, but not rename it 😕

Reply:
Oh wait, it was just a bug in reusing the `update-teams` code (trying to parse and reinsert the creator's name into the group name). It looks like an admin can rename the group after all. 👍

Reply:
Ah! Good news!